casdk-docs/blog/2024-05-16-dotnet8/index.md

```diff
@@ -0,0 +1,156 @@
+
+![fig1](./fig1.png)
+
+Fig.1 Property window of C# project in Carbon Aware SDK on Visual Studio Community Edition
+
+Carbon Aware SDK includes 30 C# projects (in v1.3.0 at least), so automation is welcomed. The target framework version is described in `/Project/PropertyGroup/TargetFramework` in `.csproj` file. For example, running the command on WSL:
+
+```
+find . -name "*.csproj" -exec sed -i 's|^\(\s\+\)<TargetFramework>net6.0</TargetFramework>$|\1<TargetFramework>net8.0</TargetFramework>|g' {} \;
+```
+
+.NET version is specified in many other places, which need to be updated as well (`grep` will list them all).
+
+* Base image in Docker file
+    * Use tag `8.0` instead of `6.0` for `mcr.microsoft.com/dotnet/sdk`
+* Tool configurations
+    * global.json
+    * dotnet-tools.json
+    * launch.json for VSCode
+* Target framework version for OpenAPI generator for .NET
+* .NET version for [actions/setup-dotnet](https://github.com/actions/setup-dotnet) in GitHub Actions Workflow
+* Comments in source files
+* Documents
+
+While the updating part is done, the work does not end there…
+
+# Unexpected work items
+
+While the .NET 8 upgrade was done, some unexpected issues surfaced. 
+
+## Ripple effect on sample code
+
+To help onboard new comers to the Carbon Aware SDK, a sample running on Azure Functions is provided.
+
+Azure Functions for .NET is transitioning one of its execution modes (the In-process model) for the Isolated worker model ([more details here](https://learn.microsoft.com/en-us/azure/azure-functions/dotnet-isolated-in-process-differences)). Moreover, .NET 8 did not provide yet an option to use the former model in its initial release (cf. [roadmap of Azure Functions](https://techcommunity.microsoft.com/t5/apps-on-azure-blog/net-on-azure-functions-august-2023-roadmap-update/ba-p/3910098)).
+
+As our sample was still implementing the in-process model (to be deprecated and not available in .NET 8 at this time), it made sense to migrate to the isolated worker model.
+
+For the code lover, there is an helpful [guide](https://learn.microsoft.com/en-us/azure/azure-functions/migrate-dotnet-to-isolated-model?tabs=net8) for the migration. This led to:
+
+* Change the version of container images to build and test
+* Update the `.csproj` file
+* Replace `Startup.cs` with `Program.cs`
+* Replace `FunctionName` attribute with `Function`
+* Change loggers to be obtained from the DI container
+* Update the document
+
+For more details browse: [Pull Request #420](https://github.com/Green-Software-Foundation/carbon-aware-sdk/pull/420).
+
+## Port Number Breaking change
+
+As Carbon Aware SDK WebAPI uses ASP.NET Core technology another collateral must do change was required since .NET 8 changed its default port number from 80 to 8080 [Microsoft Learn document](https://learn.microsoft.com/en-us/dotnet/core/compatibility/containers/8.0/aspnet-port)).
```

Comment:
```suggestion
As the Carbon Aware SDK WebAPI uses ASP.NET Core technology, another collateral change was required since .NET 8 changed its default port number from 80 to 8080 [Microsoft Learn document](https://learn.microsoft.com/en-us/dotnet/core/compatibility/containers/8.0/aspnet-port)).
```